<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>TUTORIAL &mdash; BioGraPy v1.0 beta documentation</title>
    <link rel="stylesheet" href="_static/sphinxdoc.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '',
        VERSION:     '1.0 beta',
        COLLAPSE_MODINDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="BioGraPy v1.0 beta documentation" href="index.html" />
 
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-1313600-7']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="modindex.html" title="Global Module Index"
             accesskey="M">modules</a> |</li>
        <li><a href="index.html">BioGraPy v1.0 beta documentation</a> &raquo;</li> 
      </ul>
    </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
            <h3><a href="index.html">Table Of Contents</a></h3>
            <ul>
<li><a class="reference external" href="">TUTORIAL</a><ul>
<li><a class="reference external" href="#basic-usage">Basic usage</a></li>
<li><a class="reference external" href="#the-panel">The Panel</a></li>
<li><a class="reference external" href="#tracks">Tracks</a><ul>
<li><a class="reference external" href="#handling-tracks">Handling Tracks</a></li>
<li><a class="reference external" href="#ordering-features-in-track">Ordering features in track</a></li>
<li><a class="reference external" href="#color-scales">Color scales</a><ul>
<li><a class="reference external" href="#colormaps">colormaps</a></li>
<li><a class="reference external" href="#max-and-min">max and min</a></li>
<li><a class="reference external" href="#colorbars">colorbars</a></li>
</ul>
</li>
</ul>
</li>
<li><a class="reference external" href="#features">Features</a><ul>
<li><a class="reference external" href="#simple-features">Simple features</a></li>
<li><a class="reference external" href="#draw-seqfeatures">Draw SeqFeatures</a><ul>
<li><a class="reference external" href="#generic">Generic</a></li>
<li><a class="reference external" href="#gene">Gene</a></li>
<li><a class="reference external" href="#segmented">Segmented</a></li>
<li><a class="reference external" href="#single-position">Single position</a></li>
</ul>
</li>
<li><a class="reference external" href="#draw-complex-features">Draw complex features</a><ul>
<li><a class="reference external" href="#mrna-and-cds">mRNA and CDS</a></li>
<li><a class="reference external" href="#transmembrane-regions">Transmembrane regions</a></li>
<li><a class="reference external" href="#secondary-structures">Secondary structures</a></li>
<li><a class="reference external" href="#domains">Domains</a></li>
</ul>
</li>
<li><a class="reference external" href="#draw-lineplots">Draw lineplots</a></li>
</ul>
</li>
<li><a class="reference external" href="#draw-a-seqrecord">Draw a SeqRecord</a></li>
<li><a class="reference external" href="#draw-a-blast-output">Draw a Blast Output</a></li>
<li><a class="reference external" href="#extending-the-library">Extending the library</a><ul>
<li><a class="reference external" href="#creating-a-custom-graphicfeature">Creating a custom GraphicFeature</a></li>
</ul>
</li>
</ul>
</li>
</ul>

            <h3>This Page</h3>
            <ul class="this-page-menu">
              <li><a href="_sources/tutorial.txt"
                     rel="nofollow">Show Source</a></li>
            </ul>
          <div id="searchbox" style="display: none">
            <h3>Quick search</h3>
              <form class="search" action="search.html" method="get">
                <input type="text" name="q" size="18" />
                <input type="submit" value="Go" />
                <input type="hidden" name="check_keywords" value="yes" />
                <input type="hidden" name="area" value="default" />
              </form>
              <p class="searchtip" style="font-size: 90%">
              Enter search terms or a module, class or function name.
              </p>
          </div>
          <script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="tutorial">
<span id="id1"></span><h1>TUTORIAL<a class="headerlink" href="#tutorial" title="Permalink to this headline">¶</a></h1>
<p>To be completed...</p>
<div class="section" id="basic-usage">
<h2>Basic usage<a class="headerlink" href="#basic-usage" title="Permalink to this headline">¶</a></h2>
<p>Initialize panel</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">panel</span><span class="o">=</span><span class="n">Panel</span><span class="p">(</span><span class="n">fig_width</span><span class="o">=</span><span class="mi">900</span><span class="p">,)</span>
</pre></div>
</div>
<p>Create a track</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">test_track</span> <span class="o">=</span> <span class="n">tracks</span><span class="o">.</span><span class="n">BaseTrack</span><span class="p">(</span><span class="n">name</span> <span class="o">=</span> <span class="s">&#39;test&#39;</span><span class="p">,</span> <span class="n">sort_by</span> <span class="o">=</span> <span class="bp">None</span><span class="p">)</span>
</pre></div>
</div>
<p>Create a feature</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">feat1</span><span class="o">=</span><span class="n">features</span><span class="o">.</span><span class="n">Simple</span><span class="p">(</span><span class="n">name</span> <span class="o">=</span> <span class="s">&#39;feat1&#39;</span><span class="p">,</span> <span class="n">start</span> <span class="o">=</span> <span class="mi">100</span><span class="p">,</span> <span class="n">end</span> <span class="o">=</span> <span class="mi">356</span><span class="p">,</span>  <span class="p">)</span>
</pre></div>
</div>
<p>Add a features to a track</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">test_track</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">feat1</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="n">test_track</span><span class="o">.</span><span class="n">extend</span><span class="p">([</span><span class="n">feat1</span><span class="p">,</span> <span class="n">feat2</span><span class="p">,</span> <span class="n">feat3</span><span class="p">])</span>
</pre></div>
</div>
<p>or directly:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">test_track</span> <span class="o">=</span> <span class="n">tracks</span><span class="o">.</span><span class="n">BaseTrack</span><span class="p">(</span><span class="n">features</span><span class="o">.</span><span class="n">Simple</span><span class="p">(</span><span class="n">name</span> <span class="o">=</span> <span class="s">&#39;feat1&#39;</span><span class="p">,</span> <span class="n">start</span> <span class="o">=</span> <span class="mi">100</span><span class="p">,</span> <span class="n">end</span> <span class="o">=</span> <span class="mi">356</span><span class="p">,</span> <span class="p">),</span>
                          <span class="n">features</span><span class="o">.</span><span class="n">Simple</span><span class="p">(</span><span class="n">name</span> <span class="o">=</span> <span class="s">&#39;feat2&#39;</span><span class="p">,</span> <span class="n">start</span> <span class="o">=</span> <span class="mi">30</span><span class="p">,</span> <span class="n">end</span> <span class="o">=</span> <span class="mi">856</span><span class="p">,</span> <span class="p">),</span>
                          <span class="n">features</span><span class="o">.</span><span class="n">Simple</span><span class="p">(</span><span class="n">name</span> <span class="o">=</span> <span class="s">&#39;feat3&#39;</span><span class="p">,</span> <span class="n">start</span> <span class="o">=</span> <span class="mi">400</span><span class="p">,</span> <span class="n">end</span> <span class="o">=</span> <span class="mi">700</span><span class="p">,</span> <span class="p">),</span>
                          <span class="n">sort_by</span> <span class="o">=</span> <span class="s">&#39;None&#39;</span><span class="p">,)</span>
</pre></div>
</div>
<p>Add a track to the panel:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">panel</span><span class="o">.</span><span class="n">add_track</span><span class="p">(</span><span class="n">test_track</span><span class="p">)</span>
</pre></div>
</div>
<p>Save image. This supports all format supported by matplotlib including:</p>
<ul class="simple">
<li>png</li>
<li>jpeg</li>
<li>pdf</li>
<li>svg</li>
</ul>
<div class="highlight-python"><div class="highlight"><pre><span class="n">panel</span><span class="o">.</span><span class="n">save</span><span class="p">(</span><span class="s">&#39;test_image.png&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>Complete basic example:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="kn">from</span> <span class="nn">biograpy</span> <span class="kn">import</span> <span class="n">Panel</span><span class="p">,</span> <span class="n">tracks</span><span class="p">,</span> <span class="n">features</span>
<span class="n">panel</span><span class="o">=</span><span class="n">Panel</span><span class="p">(</span><span class="n">fig_width</span><span class="o">=</span><span class="mi">900</span><span class="p">,</span> <span class="n">padding</span> <span class="o">=</span> <span class="mi">25</span><span class="p">)</span><span class="c">#initialize a drawer, and add some padding</span>
<span class="n">test_track</span> <span class="o">=</span> <span class="n">tracks</span><span class="o">.</span><span class="n">BaseTrack</span><span class="p">(</span><span class="n">features</span><span class="o">.</span><span class="n">Simple</span><span class="p">(</span><span class="n">name</span> <span class="o">=</span> <span class="s">&#39;feat1&#39;</span><span class="p">,</span> <span class="n">start</span> <span class="o">=</span> <span class="mi">100</span><span class="p">,</span> <span class="n">end</span> <span class="o">=</span> <span class="mi">356</span><span class="p">,</span> <span class="p">),</span>
                              <span class="n">features</span><span class="o">.</span><span class="n">Simple</span><span class="p">(</span><span class="n">name</span> <span class="o">=</span> <span class="s">&#39;feat2&#39;</span><span class="p">,</span> <span class="n">start</span> <span class="o">=</span> <span class="mi">30</span><span class="p">,</span> <span class="n">end</span> <span class="o">=</span> <span class="mi">856</span><span class="p">,</span> <span class="p">),</span>
                              <span class="n">features</span><span class="o">.</span><span class="n">Simple</span><span class="p">(</span><span class="n">name</span> <span class="o">=</span> <span class="s">&#39;feat3&#39;</span><span class="p">,</span> <span class="n">start</span> <span class="o">=</span> <span class="mi">400</span><span class="p">,</span> <span class="n">end</span> <span class="o">=</span> <span class="mi">700</span><span class="p">,</span> <span class="p">),</span>
                              <span class="n">sort_by</span> <span class="o">=</span> <span class="s">&#39;None&#39;</span><span class="p">,)</span>
<span class="n">panel</span><span class="o">.</span><span class="n">add_track</span><span class="p">(</span><span class="n">test_track</span><span class="p">)</span>
<span class="n">panel</span><span class="o">.</span><span class="n">save</span><span class="p">(</span><span class="s">&#39;basic_test.png&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>And this is the resulting image:</p>
<img alt="_images/basic_test.png" src="_images/basic_test.png" />
</div>
<div class="section" id="the-panel">
<h2>The Panel<a class="headerlink" href="#the-panel" title="Permalink to this headline">¶</a></h2>
<p>The panel object is the &#8220;canvas&#8221; to which all the objects will be drawn.</p>
<p><strong>panel.fig</strong> is the matplotlib Figure object</p>
<p><strong>panel.tracks</strong> is the collector of all the tracks. It is a Python list object</p>
<p><strong>panel.save(filepath)</strong> calling this method will draw all the tracks and features  to the panel.fig object, and will save the figure to the filepath passed</p>
<p><strong>panel.htmlmap</strong> after <tt class="docutils literal"><span class="pre">panel.save(filepath)</span></tt> has been called the image htmlmap is accessible as this panel property if the specified format was png or jpg.</p>
<p><cite>TODO</cite></p>
</div>
<div class="section" id="tracks">
<h2>Tracks<a class="headerlink" href="#tracks" title="Permalink to this headline">¶</a></h2>
<div class="section" id="handling-tracks">
<h3>Handling Tracks<a class="headerlink" href="#handling-tracks" title="Permalink to this headline">¶</a></h3>
<p>Tracks can be seen as ordered collection of features.</p>
<p>For convenience <em>track.append()</em> and <em>track.extend</em> works as the common list methods.</p>
<p><cite>TODO</cite></p>
</div>
<div class="section" id="ordering-features-in-track">
<h3>Ordering features in track<a class="headerlink" href="#ordering-features-in-track" title="Permalink to this headline">¶</a></h3>
<p>collapse (default)</p>
<p>order as they come = user defined</p>
<p>order by lenght</p>
<p>order by score</p>
<p><cite>TODO</cite></p>
</div>
<div class="section" id="color-scales">
<h3>Color scales<a class="headerlink" href="#color-scales" title="Permalink to this headline">¶</a></h3>
<div class="section" id="colormaps">
<h4>colormaps<a class="headerlink" href="#colormaps" title="Permalink to this headline">¶</a></h4>
<p><cite>TODO</cite></p>
</div>
<div class="section" id="max-and-min">
<h4>max and min<a class="headerlink" href="#max-and-min" title="Permalink to this headline">¶</a></h4>
<p><cite>TODO</cite></p>
</div>
<div class="section" id="colorbars">
<h4>colorbars<a class="headerlink" href="#colorbars" title="Permalink to this headline">¶</a></h4>
<p><cite>TODO</cite></p>
</div>
</div>
</div>
<div class="section" id="features">
<h2>Features<a class="headerlink" href="#features" title="Permalink to this headline">¶</a></h2>
<div class="section" id="simple-features">
<h3>Simple features<a class="headerlink" href="#simple-features" title="Permalink to this headline">¶</a></h3>
<p><cite>TODO</cite></p>
</div>
<div class="section" id="draw-seqfeatures">
<h3>Draw SeqFeatures<a class="headerlink" href="#draw-seqfeatures" title="Permalink to this headline">¶</a></h3>
<div class="section" id="generic">
<h4>Generic<a class="headerlink" href="#generic" title="Permalink to this headline">¶</a></h4>
<p><cite>TODO</cite></p>
</div>
<div class="section" id="gene">
<h4>Gene<a class="headerlink" href="#gene" title="Permalink to this headline">¶</a></h4>
<p><cite>TODO</cite></p>
</div>
<div class="section" id="segmented">
<h4>Segmented<a class="headerlink" href="#segmented" title="Permalink to this headline">¶</a></h4>
<p><cite>TODO</cite></p>
</div>
<div class="section" id="single-position">
<h4>Single position<a class="headerlink" href="#single-position" title="Permalink to this headline">¶</a></h4>
<p><cite>TODO</cite></p>
</div>
</div>
<div class="section" id="draw-complex-features">
<h3>Draw complex features<a class="headerlink" href="#draw-complex-features" title="Permalink to this headline">¶</a></h3>
<div class="section" id="mrna-and-cds">
<h4>mRNA and CDS<a class="headerlink" href="#mrna-and-cds" title="Permalink to this headline">¶</a></h4>
<p><cite>TODO</cite></p>
</div>
<div class="section" id="transmembrane-regions">
<h4>Transmembrane regions<a class="headerlink" href="#transmembrane-regions" title="Permalink to this headline">¶</a></h4>
<p><cite>TODO</cite></p>
</div>
<div class="section" id="secondary-structures">
<h4>Secondary structures<a class="headerlink" href="#secondary-structures" title="Permalink to this headline">¶</a></h4>
<p><cite>TODO</cite></p>
</div>
<div class="section" id="domains">
<h4>Domains<a class="headerlink" href="#domains" title="Permalink to this headline">¶</a></h4>
<div class="highlight-python"><div class="highlight"><pre><span class="kn">from</span> <span class="nn">biograpy</span> <span class="kn">import</span> <span class="n">Panel</span><span class="p">,</span> <span class="n">tracks</span><span class="p">,</span> <span class="n">features</span>
<span class="kn">from</span> <span class="nn">Bio</span> <span class="kn">import</span> <span class="n">SeqFeature</span>

<span class="n">panel</span> <span class="o">=</span> <span class="n">Panel</span><span class="p">(</span><span class="n">fig_width</span><span class="o">=</span><span class="mi">900</span><span class="p">)</span>

<span class="n">test_track</span> <span class="o">=</span> <span class="n">tracks</span><span class="o">.</span><span class="n">BaseTrack</span><span class="p">(</span><span class="n">name</span> <span class="o">=</span> <span class="s">&#39;domains&#39;</span><span class="p">,</span> <span class="n">sort_by</span> <span class="o">=</span> <span class="bp">None</span><span class="p">,</span> <span class="n">cm</span> <span class="o">=</span> <span class="s">&#39;Set3&#39;</span><span class="p">)</span>

<span class="n">feat</span> <span class="o">=</span> <span class="n">SeqFeature</span><span class="o">.</span><span class="n">SeqFeature</span><span class="p">()</span>
<span class="n">feat</span><span class="o">.</span><span class="n">location</span> <span class="o">=</span> <span class="n">SeqFeature</span><span class="o">.</span><span class="n">FeatureLocation</span><span class="p">(</span><span class="mi">100</span><span class="p">,</span> <span class="mi">300</span><span class="p">)</span>

<span class="n">dfeat</span> <span class="o">=</span> <span class="n">features</span><span class="o">.</span><span class="n">DomainFeature</span><span class="p">([</span><span class="n">feat</span><span class="p">],</span> <span class="n">name</span> <span class="o">=</span> <span class="s">&#39;test domain 1&#39;</span><span class="p">,</span> <span class="n">seq_line</span> <span class="o">=</span> <span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">766</span><span class="p">))</span>
<span class="n">test_track</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">dfeat</span><span class="p">)</span>

<span class="n">dfeat</span> <span class="o">=</span> <span class="n">features</span><span class="o">.</span><span class="n">DomainFeature</span><span class="p">([</span><span class="n">feat</span><span class="p">],</span><span class="n">name</span> <span class="o">=</span> <span class="s">&#39;test domain 1&#39;</span><span class="p">,</span> <span class="n">height</span> <span class="o">=</span> <span class="mf">1.5</span><span class="p">,</span> <span class="n">seq_line</span> <span class="o">=</span> <span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">766</span><span class="p">))</span>
<span class="n">test_track</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">dfeat</span><span class="p">)</span>

<span class="n">dfeat</span> <span class="o">=</span> <span class="n">features</span><span class="o">.</span><span class="n">DomainFeature</span><span class="p">([</span><span class="n">feat</span><span class="p">],</span><span class="n">name</span> <span class="o">=</span> <span class="s">&#39;test domain 1&#39;</span><span class="p">,</span> <span class="n">height</span> <span class="o">=</span> <span class="mf">1.5</span><span class="p">,</span> <span class="n">boxstyle</span> <span class="o">=</span> <span class="s">&#39;round4, rounding_size=1.4&#39;</span><span class="p">,</span>  <span class="n">seq_line</span> <span class="o">=</span> <span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">766</span><span class="p">))</span>
<span class="n">test_track</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">dfeat</span><span class="p">)</span>

<span class="n">dfeat</span> <span class="o">=</span> <span class="n">features</span><span class="o">.</span><span class="n">DomainFeature</span><span class="p">([</span><span class="n">feat</span><span class="p">],</span><span class="n">name</span> <span class="o">=</span> <span class="s">&#39;test domain 1&#39;</span><span class="p">,</span><span class="n">height</span> <span class="o">=</span> <span class="mf">1.5</span><span class="p">,</span> <span class="n">boxstyle</span> <span class="o">=</span> <span class="s">&#39;larrow, pad = 0&#39;</span><span class="p">,</span>  <span class="n">seq_line</span> <span class="o">=</span> <span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">766</span><span class="p">))</span>
<span class="n">test_track</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">dfeat</span><span class="p">)</span>

<span class="n">dfeat</span> <span class="o">=</span> <span class="n">features</span><span class="o">.</span><span class="n">DomainFeature</span><span class="p">([</span><span class="n">feat</span><span class="p">],</span> <span class="n">name</span> <span class="o">=</span> <span class="s">&#39;test domain 1&#39;</span><span class="p">,</span><span class="n">height</span> <span class="o">=</span> <span class="mf">1.5</span><span class="p">,</span> <span class="n">boxstyle</span> <span class="o">=</span> <span class="s">&#39;round &#39;</span><span class="p">,</span>  <span class="n">seq_line</span> <span class="o">=</span> <span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">766</span><span class="p">))</span>
<span class="n">test_track</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">dfeat</span><span class="p">)</span>

<span class="n">feat2</span> <span class="o">=</span> <span class="n">SeqFeature</span><span class="o">.</span><span class="n">SeqFeature</span><span class="p">()</span>
<span class="n">feat2</span><span class="o">.</span><span class="n">location</span> <span class="o">=</span> <span class="n">SeqFeature</span><span class="o">.</span><span class="n">FeatureLocation</span><span class="p">(</span><span class="mi">500</span><span class="p">,</span> <span class="mi">700</span><span class="p">)</span>

<span class="n">dfeat</span> <span class="o">=</span> <span class="n">features</span><span class="o">.</span><span class="n">DomainFeature</span><span class="p">([</span><span class="n">feat</span><span class="p">,</span><span class="n">feat2</span><span class="p">],</span> <span class="n">name</span> <span class="o">=</span> <span class="s">&#39;test domain 1&#39;</span><span class="p">,</span><span class="n">height</span> <span class="o">=</span> <span class="mf">1.5</span><span class="p">,</span> <span class="n">boxstyle</span> <span class="o">=</span> <span class="s">&#39;roundtooth, pad = 0.1, tooth_size=1.2&#39;</span><span class="p">,</span>  <span class="n">seq_line</span> <span class="o">=</span> <span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">766</span><span class="p">),</span> <span class="n">alpha</span> <span class="o">=</span> <span class="mf">0.7</span><span class="p">)</span>
<span class="n">test_track</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">dfeat</span><span class="p">)</span>

<span class="n">dfeat</span> <span class="o">=</span> <span class="n">features</span><span class="o">.</span><span class="n">DomainFeature</span><span class="p">([</span><span class="n">feat</span><span class="p">,</span><span class="n">feat2</span><span class="p">],</span><span class="n">name</span> <span class="o">=</span> <span class="p">[</span><span class="s">&#39;test domain 1&#39;</span><span class="p">,</span> <span class="s">&#39;test domain 2&#39;</span><span class="p">],</span> <span class="n">height</span> <span class="o">=</span> <span class="mf">1.5</span><span class="p">,</span> <span class="n">boxstyle</span> <span class="o">=</span> <span class="p">[</span><span class="s">&#39;sawtooth, tooth_size=1.4&#39;</span><span class="p">,</span>  <span class="s">&#39;rarrow, pad = 0.1&#39;</span><span class="p">],</span>  <span class="n">seq_line</span> <span class="o">=</span> <span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">766</span><span class="p">),</span> <span class="n">ec</span> <span class="o">=</span> <span class="s">&#39;None&#39;</span><span class="p">)</span>
<span class="n">test_track</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">dfeat</span><span class="p">)</span>

<span class="n">dfeat</span> <span class="o">=</span> <span class="n">features</span><span class="o">.</span><span class="n">DomainFeature</span><span class="p">([</span><span class="n">feat</span><span class="p">,</span><span class="n">feat2</span><span class="p">],</span><span class="n">name</span> <span class="o">=</span> <span class="p">[</span><span class="s">&#39;&#39;</span><span class="p">,</span> <span class="s">&#39;test domain 2&#39;</span><span class="p">],</span><span class="n">height</span> <span class="o">=</span> <span class="mf">1.5</span><span class="p">,</span> <span class="n">boxstyle</span> <span class="o">=</span> <span class="p">[</span><span class="s">&#39;round4, rounding_size=1.4&#39;</span><span class="p">,</span>  <span class="s">&#39;larrow, pad = 0.1&#39;</span><span class="p">],</span>   <span class="n">ec</span> <span class="o">=</span> <span class="s">&#39;None&#39;</span><span class="p">,</span> <span class="n">fc</span> <span class="o">=</span> <span class="p">[</span><span class="s">&#39;y&#39;</span><span class="p">,</span> <span class="s">&#39;c&#39;</span><span class="p">],</span> <span class="n">color_by_cm</span> <span class="o">=</span> <span class="bp">False</span><span class="p">)</span>
<span class="n">test_track</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">dfeat</span><span class="p">)</span>

<span class="n">panel</span><span class="o">.</span><span class="n">add_track</span><span class="p">(</span><span class="n">test_track</span><span class="p">)</span>
<span class="n">panel</span><span class="o">.</span><span class="n">save</span><span class="p">(</span><span class="s">&#39;domain_test.png&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>resulting image:</p>
<img alt="_images/domain_test.png" src="_images/domain_test.png" />
</div>
</div>
<div class="section" id="draw-lineplots">
<h3>Draw lineplots<a class="headerlink" href="#draw-lineplots" title="Permalink to this headline">¶</a></h3>
<p>Plot feature requires plot track</p>
<p><cite>TODO</cite></p>
</div>
</div>
<div class="section" id="draw-a-seqrecord">
<h2>Draw a SeqRecord<a class="headerlink" href="#draw-a-seqrecord" title="Permalink to this headline">¶</a></h2>
<p>Starting from <a class="reference external" href="https://github.com/apierleoni/BioGraPy/blob/master/src/biograpy/tests/factor7.embl">this</a> annotated entry in EMBL format</p>
<p>Generate a plot</p>
<div class="highlight-python"><div class="highlight"><pre><span class="kn">from</span> <span class="nn">biograpy.seqrecord</span> <span class="kn">import</span> <span class="n">SeqRecordDrawer</span>
<span class="kn">from</span> <span class="nn">biograpy</span> <span class="kn">import</span> <span class="n">Panel</span><span class="p">,</span> <span class="n">tracks</span><span class="p">,</span> <span class="n">features</span>
<span class="kn">from</span> <span class="nn">Bio</span> <span class="kn">import</span> <span class="n">SeqIO</span>
<span class="n">seqrecord</span> <span class="o">=</span> <span class="n">SeqIO</span><span class="o">.</span><span class="n">parse</span><span class="p">(</span><span class="nb">open</span><span class="p">(</span><span class="s">&quot;factor7.embl&quot;</span><span class="p">),</span> <span class="s">&quot;embl&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">next</span><span class="p">()</span>
<span class="n">SeqRecordDrawer</span><span class="p">(</span><span class="n">seqrecord</span><span class="p">,</span> <span class="n">fig_width</span><span class="o">=</span><span class="mi">900</span><span class="p">)</span><span class="o">.</span><span class="n">save</span><span class="p">(</span><span class="s">&quot;factor7.png&quot;</span><span class="p">)</span>
</pre></div>
</div>
<img alt="_images/factor7.png" src="_images/factor7.png" />
<p>Check out the similar plot done with BioPerl <a class="reference external" href="http://www.bioperl.org/wiki/HOWTO:Graphics#Rendering_Features_from_a_GenBank_or_EMBL_File">here</a>.</p>
</div>
<div class="section" id="draw-a-blast-output">
<h2>Draw a Blast Output<a class="headerlink" href="#draw-a-blast-output" title="Permalink to this headline">¶</a></h2>
<div class="highlight-python"><div class="highlight"><pre><span class="kn">from</span> <span class="nn">biograpy</span> <span class="kn">import</span> <span class="n">Panel</span><span class="p">,</span> <span class="n">tracks</span><span class="p">,</span> <span class="n">features</span>
<span class="kn">from</span> <span class="nn">Bio</span> <span class="kn">import</span> <span class="n">SeqIO</span><span class="p">,</span> <span class="n">AlignIO</span><span class="p">,</span> <span class="n">SeqFeature</span>
<span class="kn">from</span> <span class="nn">Bio.Blast</span> <span class="kn">import</span> <span class="n">NCBIXML</span>

<span class="sd">&#39;&#39;&#39;Parse BLAST XML output&#39;&#39;&#39;</span>
<span class="n">aligndata</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">blasthits</span><span class="o">=</span><span class="n">NCBIXML</span><span class="o">.</span><span class="n">parse</span><span class="p">(</span><span class="nb">open</span><span class="p">(</span><span class="s">&#39;BLASTalignment.xml&#39;</span><span class="p">))</span>
<span class="n">id_filter</span> <span class="o">=</span> <span class="o">.</span><span class="mi">3</span> <span class="c">#show only alignments with sequence identity higher than 30%</span>
<span class="k">for</span> <span class="n">hit</span> <span class="ow">in</span> <span class="n">blasthits</span><span class="p">:</span>
    <span class="k">for</span> <span class="n">alignment</span> <span class="ow">in</span> <span class="n">hit</span><span class="o">.</span><span class="n">alignments</span><span class="p">:</span>
        <span class="n">hsps</span> <span class="o">=</span> <span class="p">[]</span>
        <span class="k">for</span> <span class="n">hsp</span> <span class="ow">in</span> <span class="n">alignment</span><span class="o">.</span><span class="n">hsps</span><span class="p">:</span>
            <span class="k">if</span> <span class="p">(</span><span class="n">hsp</span><span class="o">.</span><span class="n">identities</span><span class="o">/</span><span class="nb">float</span><span class="p">(</span><span class="n">hsp</span><span class="o">.</span><span class="n">align_length</span><span class="p">)</span> <span class="o">&gt;=</span> <span class="n">id_filter</span><span class="p">):</span>
                <span class="n">hit_id</span> <span class="o">=</span> <span class="n">alignment</span><span class="o">.</span><span class="n">hit_id</span>
                <span class="n">hit_name</span> <span class="o">=</span> <span class="n">alignment</span><span class="o">.</span><span class="n">hit_def</span> <span class="p">[:</span><span class="mi">40</span><span class="p">]</span>
                <span class="n">align</span><span class="o">=</span><span class="nb">dict</span><span class="p">(</span><span class="n">hit_id</span> <span class="o">=</span> <span class="n">hit_id</span><span class="p">,</span>
                           <span class="n">hit_name</span> <span class="o">=</span> <span class="n">hit_name</span><span class="p">,</span>
                           <span class="n">hit_start</span> <span class="o">=</span> <span class="n">hsp</span><span class="o">.</span><span class="n">sbjct_start</span><span class="p">,</span>
                           <span class="n">hit_stop</span> <span class="o">=</span> <span class="n">hsp</span><span class="o">.</span><span class="n">sbjct_end</span><span class="p">,</span>
                           <span class="n">query_start</span> <span class="o">=</span> <span class="n">hsp</span><span class="o">.</span><span class="n">query_start</span><span class="p">,</span>
                           <span class="n">query_stop</span> <span class="o">=</span> <span class="n">hsp</span><span class="o">.</span><span class="n">query_end</span><span class="p">,</span>
                           <span class="n">align_lenght</span> <span class="o">=</span> <span class="n">hsp</span><span class="o">.</span><span class="n">align_length</span><span class="p">,</span>
                           <span class="n">identity_percent</span> <span class="o">=</span> <span class="n">hsp</span><span class="o">.</span><span class="n">identities</span><span class="o">/</span><span class="nb">float</span><span class="p">(</span><span class="n">hsp</span><span class="o">.</span><span class="n">align_length</span><span class="p">)</span><span class="o">*</span><span class="mi">100</span><span class="p">,</span>
                           <span class="n">similarity_percent</span> <span class="o">=</span> <span class="n">hsp</span><span class="o">.</span><span class="n">positives</span><span class="o">/</span><span class="nb">float</span><span class="p">(</span><span class="n">hsp</span><span class="o">.</span><span class="n">align_length</span><span class="p">)</span><span class="o">*</span><span class="mi">100</span><span class="p">,</span>
                           <span class="n">evalue</span> <span class="o">=</span> <span class="n">hsp</span><span class="o">.</span><span class="n">expect</span><span class="p">,</span>
                           <span class="n">blast_score</span> <span class="o">=</span> <span class="n">hsp</span><span class="o">.</span><span class="n">score</span><span class="p">,</span>
                           <span class="n">query_seq</span> <span class="o">=</span> <span class="n">hsp</span><span class="o">.</span><span class="n">query</span><span class="p">,</span>
                           <span class="n">match_seq</span> <span class="o">=</span> <span class="n">hsp</span><span class="o">.</span><span class="n">match</span><span class="p">,</span>
                           <span class="n">hit_seq</span> <span class="o">=</span> <span class="n">hsp</span><span class="o">.</span><span class="n">sbjct</span><span class="p">,</span>
                           <span class="n">aligns</span> <span class="o">=</span> <span class="n">hsp</span><span class="o">.</span><span class="n">num_alignments</span><span class="p">,</span>
                           <span class="n">other</span> <span class="o">=</span> <span class="nb">dir</span><span class="p">(</span><span class="n">hsp</span><span class="p">))</span>
                <span class="n">hsps</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">align</span><span class="p">)</span>


        <span class="n">aligndata</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="nb">dict</span><span class="p">(</span><span class="n">hsps</span><span class="o">=</span> <span class="n">hsps</span><span class="p">,</span>
                           <span class="n">hit_def</span> <span class="o">=</span> <span class="n">alignment</span><span class="o">.</span><span class="n">hit_def</span><span class="p">,</span>
                           <span class="n">hit_id</span> <span class="o">=</span> <span class="n">alignment</span><span class="o">.</span><span class="n">hit_id</span><span class="p">,</span>
                           <span class="n">length</span> <span class="o">=</span> <span class="n">alignment</span><span class="o">.</span><span class="n">length</span><span class="p">,</span>
                           <span class="n">title</span> <span class="o">=</span> <span class="n">alignment</span><span class="o">.</span><span class="n">title</span><span class="p">,))</span>

<span class="sd">&#39;&#39;&#39;Draw Alignment&#39;&#39;&#39;</span>
<span class="n">panel</span><span class="o">=</span><span class="n">Panel</span><span class="p">(</span><span class="n">fig_width</span><span class="o">=</span><span class="mi">900</span><span class="p">,)</span><span class="c">#initialize a drawer</span>
<span class="n">ref_track</span> <span class="o">=</span> <span class="n">tracks</span><span class="o">.</span><span class="n">BaseTrack</span><span class="p">(</span><span class="n">features</span><span class="o">.</span><span class="n">Simple</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span><span class="c">#draw a reference object</span>
                                           <span class="n">hit</span><span class="o">.</span><span class="n">query_length</span><span class="p">,</span>
                                           <span class="n">name</span> <span class="o">=</span> <span class="n">hit</span><span class="o">.</span><span class="n">query</span><span class="p">,</span>
                                           <span class="n">height</span> <span class="o">=</span> <span class="mf">1.5</span><span class="p">,</span>
                                           <span class="n">fc</span><span class="o">=</span><span class="s">&#39;grey&#39;</span><span class="p">,</span>
                                           <span class="n">ec</span><span class="o">=</span><span class="s">&#39;k&#39;</span><span class="p">,),</span>
                             <span class="n">name</span> <span class="o">=</span> <span class="s">&#39;Query&#39;</span><span class="p">)</span>
<span class="n">panel</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">ref_track</span><span class="p">)</span>
<span class="n">aligns_track</span> <span class="o">=</span> <span class="n">tracks</span><span class="o">.</span><span class="n">BaseTrack</span><span class="p">(</span><span class="n">name</span> <span class="o">=</span> <span class="s">&#39;Hits&#39;</span><span class="p">,</span> <span class="n">cm</span> <span class="o">=</span><span class="s">&#39;Blues&#39;</span><span class="p">,</span> <span class="p">)</span>
<span class="k">for</span> <span class="n">align</span> <span class="ow">in</span> <span class="n">aligndata</span><span class="p">:</span>
    <span class="k">for</span> <span class="n">hsp</span> <span class="ow">in</span>  <span class="n">align</span><span class="p">[</span><span class="s">&#39;hsps&#39;</span><span class="p">]:</span>
        <span class="n">location</span> <span class="o">=</span> <span class="n">SeqFeature</span><span class="o">.</span><span class="n">FeatureLocation</span><span class="p">(</span><span class="n">hsp</span><span class="p">[</span><span class="s">&#39;query_start&#39;</span><span class="p">],</span> <span class="n">hsp</span><span class="p">[</span><span class="s">&#39;query_stop&#39;</span><span class="p">])</span>
        <span class="n">feat</span> <span class="o">=</span> <span class="n">SeqFeature</span><span class="o">.</span><span class="n">SeqFeature</span><span class="p">()</span>
        <span class="n">feat</span><span class="o">.</span><span class="n">type</span> <span class="o">=</span> <span class="s">&#39;Hit&#39;</span>
        <span class="n">feat</span><span class="o">.</span><span class="n">id</span> <span class="o">=</span> <span class="n">hsp</span><span class="p">[</span><span class="s">&#39;hit_id&#39;</span><span class="p">]</span>
        <span class="n">feat</span><span class="o">.</span><span class="n">location</span> <span class="o">=</span> <span class="n">location</span>
        <span class="n">feat</span><span class="o">.</span><span class="n">qualifiers</span><span class="p">[</span><span class="s">&#39;ft_description&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="s">&#39;Hit&#39;</span><span class="p">]</span>
        <span class="n">feat</span><span class="o">.</span><span class="n">qualifiers</span><span class="p">[</span><span class="s">&#39;ft_type&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="s">&#39;Hit&#39;</span><span class="p">]</span>
        <span class="n">feat</span><span class="o">.</span><span class="n">qualifiers</span><span class="p">[</span><span class="s">&#39;score&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="nb">float</span><span class="p">(</span><span class="n">hsp</span><span class="p">[</span><span class="s">&#39;identity_percent&#39;</span><span class="p">])</span><span class="o">/</span><span class="mf">100.</span>
        <span class="n">feat</span><span class="o">.</span><span class="n">qualifiers</span><span class="p">[</span><span class="s">&#39;hit_name&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="n">hsp</span><span class="p">[</span><span class="s">&#39;hit_id&#39;</span><span class="p">]]</span>
        <span class="n">aligns_track</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">features</span><span class="o">.</span><span class="n">GenericSeqFeature</span><span class="p">(</span><span class="n">feat</span><span class="p">,</span>
                                           <span class="n">score</span> <span class="o">=</span> <span class="n">feat</span><span class="o">.</span><span class="n">qualifiers</span><span class="p">[</span><span class="s">&#39;score&#39;</span><span class="p">],</span>
                                           <span class="n">name</span> <span class="o">=</span> <span class="n">hsp</span><span class="p">[</span><span class="s">&#39;hit_name&#39;</span><span class="p">],</span>
                                           <span class="n">use_score_for_color</span> <span class="o">=</span> <span class="bp">True</span><span class="p">,</span>
                                           <span class="n">url</span> <span class="o">=</span> <span class="s">&quot;http://www.ncbi.nlm.nih.gov/protein/</span><span class="si">%s</span><span class="s">&quot;</span><span class="o">%</span><span class="n">hsp</span><span class="p">[</span><span class="s">&#39;hit_id&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s">&#39;|&#39;</span><span class="p">)[</span><span class="o">-</span><span class="mi">2</span><span class="p">],))</span>

<span class="n">panel</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">aligns_track</span><span class="p">)</span>


<span class="n">panel</span><span class="o">.</span><span class="n">save</span><span class="p">(</span><span class="s">&#39;blast_output.png&#39;</span><span class="p">)</span>
<span class="k">print</span> <span class="n">panel</span><span class="o">.</span><span class="n">htmlmap</span>
</pre></div>
</div>
<p>Generated image:</p>
<img alt="_images/blast_output.png" src="_images/blast_output.png" />
<p>to be coupled with the generated html map available calling</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">panel</span><span class="o">.</span><span class="n">htmlmap</span>
</pre></div>
</div>
<div class="highlight-python"><pre>&lt;map name="biograpy-map" id="biograpy-map"&gt;
 &lt;area shape="rect" coords="18,43,882,28" href="#sp|P04626|ERBB2_HUMAN Receptor tyrosine-protein kinase erbB-2 OS=Homo sapiens GN=ERBB2 PE=1 SV=1" target="_self" alt="sp|P04626|ERBB2_HUMAN Receptor tyrosine-protein kinase erbB-2 OS=Homo sapiens GN=ERBB2 PE=1 SV=1"  &gt;
&lt;area shape="rect" coords="18,93,882,81" href="http://www.ncbi.nlm.nih.gov/protein/NP_004439.2" target="_self" alt="receptor tyrosine-protein kinase erbB-2 "  &gt;
&lt;area shape="rect" coords="38,116,882,105" href="http://www.ncbi.nlm.nih.gov/protein/NP_001005862.1" target="_self" alt="receptor tyrosine-protein kinase erbB-2 "  &gt;
&lt;area shape="rect" coords="33,139,877,128" href="http://www.ncbi.nlm.nih.gov/protein/NP_005219.2" target="_self" alt="epidermal growth factor receptor isoform"  &gt;
&lt;area shape="rect" coords="23,163,877,151" href="http://www.ncbi.nlm.nih.gov/protein/NP_001036064.1" target="_self" alt="receptor tyrosine-protein kinase erbB-4 "  &gt;
&lt;area shape="rect" coords="32,186,877,174" href="http://www.ncbi.nlm.nih.gov/protein/NP_005226.1" target="_self" alt="receptor tyrosine-protein kinase erbB-4 "  &gt;
&lt;area shape="rect" coords="20,209,704,197" href="http://www.ncbi.nlm.nih.gov/protein/NP_001973.2" target="_self" alt="receptor tyrosine-protein kinase erbB-3 "  &gt;
&lt;area shape="rect" coords="33,232,452,221" href="http://www.ncbi.nlm.nih.gov/protein/NP_958439.1" target="_self" alt="epidermal growth factor receptor isoform"  &gt;
&lt;area shape="rect" coords="33,255,452,244" href="http://www.ncbi.nlm.nih.gov/protein/NP_958441.1" target="_self" alt="epidermal growth factor receptor isoform"  &gt;
&lt;area shape="rect" coords="33,279,298,267" href="http://www.ncbi.nlm.nih.gov/protein/NP_958440.1" target="_self" alt="epidermal growth factor receptor isoform"  &gt;
&lt;area shape="rect" coords="432,279,706,267" href="http://www.ncbi.nlm.nih.gov/protein/NP_004432.1" target="_self" alt="ephrin type-B receptor 1 precursor [Homo"  &gt;
&lt;area shape="rect" coords="479,232,701,221" href="http://www.ncbi.nlm.nih.gov/protein/NP_004422.2" target="_self" alt="ephrin type-A receptor 2 precursor [Homo"  &gt;
&lt;area shape="rect" coords="509,255,693,244" href="http://www.ncbi.nlm.nih.gov/protein/NP_001129473.1" target="_self" alt="tyrosine-protein kinase ABL2 isoform e ["  &gt;
&lt;area shape="rect" coords="461,302,716,290" href="http://www.ncbi.nlm.nih.gov/protein/NP_872272.2" target="_self" alt="ephrin type-A receptor 5 isoform b precu"  &gt;
&lt;area shape="rect" coords="504,325,706,313" href="http://www.ncbi.nlm.nih.gov/protein/NP_004435.3" target="_self" alt="ephrin type-B receptor 4 precursor [Homo"  &gt;
&lt;area shape="rect" coords="504,348,716,336" href="http://www.ncbi.nlm.nih.gov/protein/NP_004430.4" target="_self" alt="ephrin type-A receptor 5 isoform a precu"  &gt;
&lt;area shape="rect" coords="516,371,693,360" href="http://www.ncbi.nlm.nih.gov/protein/NP_004094.3" target="_self" alt="protein-tyrosine kinase 2-beta isoform a"  &gt;
&lt;area shape="rect" coords="516,394,693,383" href="http://www.ncbi.nlm.nih.gov/protein/NP_775267.1" target="_self" alt="protein-tyrosine kinase 2-beta isoform b"  &gt;
&lt;area shape="rect" coords="467,418,695,406" href="http://www.ncbi.nlm.nih.gov/protein/NP_004433.2" target="_self" alt="ephrin type-B receptor 2 isoform 2 precu"  &gt;
&lt;area shape="rect" coords="467,441,695,429" href="http://www.ncbi.nlm.nih.gov/protein/NP_059145.2" target="_self" alt="ephrin type-B receptor 2 isoform 1 precu"  &gt;
&lt;area shape="rect" coords="509,464,693,452" href="http://www.ncbi.nlm.nih.gov/protein/NP_001129472.1" target="_self" alt="tyrosine-protein kinase ABL2 isoform d ["  &gt;
&lt;area shape="rect" coords="509,487,693,476" href="http://www.ncbi.nlm.nih.gov/protein/NP_001161711.1" target="_self" alt="tyrosine-protein kinase ABL2 isoform i ["  &gt;
&lt;area shape="rect" coords="509,510,693,499" href="http://www.ncbi.nlm.nih.gov/protein/NP_005149.4" target="_self" alt="tyrosine-protein kinase ABL2 isoform c ["  &gt;
&lt;area shape="rect" coords="509,534,693,522" href="http://www.ncbi.nlm.nih.gov/protein/NP_001161708.1" target="_self" alt="tyrosine-protein kinase ABL2 isoform f ["  &gt;
&lt;area shape="rect" coords="509,557,693,545" href="http://www.ncbi.nlm.nih.gov/protein/NP_009298.1" target="_self" alt="tyrosine-protein kinase ABL2 isoform b ["  &gt;
&lt;area shape="rect" coords="509,580,693,568" href="http://www.ncbi.nlm.nih.gov/protein/NP_001161709.1" target="_self" alt="tyrosine-protein kinase ABL2 isoform g ["  &gt;
&lt;area shape="rect" coords="509,603,693,592" href="http://www.ncbi.nlm.nih.gov/protein/NP_001161710.1" target="_self" alt="tyrosine-protein kinase ABL2 isoform h ["  &gt;
&lt;area shape="rect" coords="503,626,690,615" href="http://www.ncbi.nlm.nih.gov/protein/NP_722560.1" target="_self" alt="focal adhesion kinase 1 isoform a [Homo "  &gt;
&lt;area shape="rect" coords="498,650,687,638" href="http://www.ncbi.nlm.nih.gov/protein/NP_005772.3" target="_self" alt="activated CDC42 kinase 1 isoform 1 [Homo"  &gt;
&lt;area shape="rect" coords="503,673,695,661" href="http://www.ncbi.nlm.nih.gov/protein/NP_005598.3" target="_self" alt="focal adhesion kinase 1 isoform b [Homo "  &gt;
&lt;area shape="rect" coords="517,696,684,684" href="http://www.ncbi.nlm.nih.gov/protein/NP_997402.1" target="_self" alt="tyrosine-protein kinase ZAP-70 isoform 2"  &gt;
&lt;area shape="rect" coords="503,719,690,708" href="http://www.ncbi.nlm.nih.gov/protein/NP_001186578.1" target="_self" alt="focal adhesion kinase 1 isoform c [Homo "  &gt;
&lt;area shape="rect" coords="498,742,687,731" href="http://www.ncbi.nlm.nih.gov/protein/NP_001010938.1" target="_self" alt="activated CDC42 kinase 1 isoform 2 [Homo"  &gt;
&lt;area shape="rect" coords="478,766,706,754" href="http://www.ncbi.nlm.nih.gov/protein/NP_004429.1" target="_self" alt="ephrin type-A receptor 4 precursor [Homo"  &gt;
&lt;area shape="rect" coords="504,789,711,777" href="http://www.ncbi.nlm.nih.gov/protein/NP_005224.2" target="_self" alt="ephrin type-A receptor 3 isoform a precu"  &gt;
&lt;area shape="rect" coords="502,812,693,800" href="http://www.ncbi.nlm.nih.gov/protein/NP_005148.2" target="_self" alt="tyrosine-protein kinase ABL1 isoform a ["  &gt;
&lt;area shape="rect" coords="502,835,693,823" href="http://www.ncbi.nlm.nih.gov/protein/NP_009297.2" target="_self" alt="tyrosine-protein kinase ABL1 isoform b ["  &gt;
&lt;area shape="rect" coords="504,858,720,847" href="http://www.ncbi.nlm.nih.gov/protein/NP_004434.2" target="_self" alt="ephrin type-B receptor 3 precursor [Homo"  &gt;
&lt;area shape="rect" coords="516,881,724,870" href="http://www.ncbi.nlm.nih.gov/protein/NP_065681.1" target="_self" alt="proto-oncogene tyrosine-protein kinase r"  &gt;
&lt;area shape="rect" coords="516,905,724,893" href="http://www.ncbi.nlm.nih.gov/protein/NP_066124.1" target="_self" alt="proto-oncogene tyrosine-protein kinase r"  &gt;
&lt;area shape="rect" coords="517,928,684,916" href="http://www.ncbi.nlm.nih.gov/protein/NP_001070.2" target="_self" alt="tyrosine-protein kinase ZAP-70 isoform 1"  &gt;
&lt;area shape="rect" coords="516,951,684,939" href="http://www.ncbi.nlm.nih.gov/protein/NP_001128524.1" target="_self" alt="tyrosine-protein kinase SYK isoform 2 [H"  &gt;
&lt;area shape="rect" coords="516,974,684,963" href="http://www.ncbi.nlm.nih.gov/protein/NP_003168.2" target="_self" alt="tyrosine-protein kinase SYK isoform 1 [H"  &gt;
&lt;area shape="rect" coords="516,997,691,986" href="http://www.ncbi.nlm.nih.gov/protein/NP_000236.2" target="_self" alt="hepatocyte growth factor receptor isofor"  &gt;
&lt;area shape="rect" coords="474,1021,706,1009" href="http://www.ncbi.nlm.nih.gov/protein/NP_004431.1" target="_self" alt="ephrin type-A receptor 7 precursor [Homo"  &gt;
&lt;area shape="rect" coords="516,1044,691,1032" href="http://www.ncbi.nlm.nih.gov/protein/NP_001120972.1" target="_self" alt="hepatocyte growth factor receptor isofor"  &gt;
&lt;area shape="rect" coords="516,1067,692,1055" href="http://www.ncbi.nlm.nih.gov/protein/NP_002438.2" target="_self" alt="macrophage-stimulating protein receptor "  &gt;
&lt;area shape="rect" coords="504,1090,701,1079" href="http://www.ncbi.nlm.nih.gov/protein/NP_005223.4" target="_self" alt="ephrin type-A receptor 1 precursor [Homo"  &gt;
&lt;area shape="rect" coords="504,1113,695,1102" href="http://www.ncbi.nlm.nih.gov/protein/NP_065387.1" target="_self" alt="ephrin type-A receptor 8 isoform 1 precu"  &gt;
&lt;area shape="rect" coords="509,1137,692,1125" href="http://www.ncbi.nlm.nih.gov/protein/NP_005537.3" target="_self" alt="tyrosine-protein kinase ITK/TSK [Homo sa"  &gt;
&lt;area shape="rect" coords="512,1160,692,1148" href="http://www.ncbi.nlm.nih.gov/protein/NP_003319.2" target="_self" alt="tyrosine-protein kinase TXK [Homo sapien"  &gt;
&lt;area shape="rect" coords="509,1183,691,1171" href="http://www.ncbi.nlm.nih.gov/protein/NP_075254.1" target="_self" alt="fibroblast growth factor receptor 3 isof"  &gt;
&lt;area shape="rect" coords="509,1206,691,1195" href="http://www.ncbi.nlm.nih.gov/protein/NP_000133.1" target="_self" alt="fibroblast growth factor receptor 3 isof"  &gt;
&lt;area shape="rect" coords="509,1229,691,1218" href="http://www.ncbi.nlm.nih.gov/protein/NP_001156685.1" target="_self" alt="fibroblast growth factor receptor 3 isof"  &gt;
&lt;area shape="rect" coords="512,1252,687,1241" href="http://www.ncbi.nlm.nih.gov/protein/NP_003206.2" target="_self" alt="tyrosine-protein kinase Tec [Homo sapien"  &gt;
&lt;area shape="rect" coords="516,1276,706,1264" href="http://www.ncbi.nlm.nih.gov/protein/NP_075594.1" target="_self" alt="basic fibroblast growth factor receptor "  &gt;
&lt;area shape="rect" coords="516,1299,706,1287" href="http://www.ncbi.nlm.nih.gov/protein/NP_075593.1" target="_self" alt="basic fibroblast growth factor receptor "  &gt;
&lt;area shape="rect" coords="509,1322,687,1310" href="http://www.ncbi.nlm.nih.gov/protein/NP_002022.1" target="_self" alt="tyrosine-protein kinase FRK [Homo sapien"  &gt;
&lt;area shape="rect" coords="516,1345,692,1334" href="http://www.ncbi.nlm.nih.gov/protein/NP_001167535.1" target="_self" alt="basic fibroblast growth factor receptor "  &gt;
&lt;area shape="rect" coords="516,1368,692,1357" href="http://www.ncbi.nlm.nih.gov/protein/NP_001167534.1" target="_self" alt="basic fibroblast growth factor receptor "  &gt;
&lt;area shape="rect" coords="516,1392,692,1380" href="http://www.ncbi.nlm.nih.gov/protein/NP_056934.2" target="_self" alt="basic fibroblast growth factor receptor "  &gt;
&lt;area shape="rect" coords="516,1415,691,1403" href="http://www.ncbi.nlm.nih.gov/protein/NP_001138391.1" target="_self" alt="fibroblast growth factor receptor 2 isof"  &gt;
&lt;area shape="rect" coords="516,1438,692,1426" href="http://www.ncbi.nlm.nih.gov/protein/NP_075598.2" target="_self" alt="basic fibroblast growth factor receptor "  &gt;
&lt;area shape="rect" coords="516,1461,691,1450" href="http://www.ncbi.nlm.nih.gov/protein/NP_001138386.1" target="_self" alt="fibroblast growth factor receptor 2 isof"  &gt;
&lt;area shape="rect" coords="516,1484,691,1473" href="http://www.ncbi.nlm.nih.gov/protein/NP_001138389.1" target="_self" alt="fibroblast growth factor receptor 2 isof"  &gt;
&lt;area shape="rect" coords="516,1508,706,1496" href="http://www.ncbi.nlm.nih.gov/protein/NP_001167538.1" target="_self" alt="basic fibroblast growth factor receptor "  &gt;
&lt;area shape="rect" coords="516,1531,691,1519" href="http://www.ncbi.nlm.nih.gov/protein/NP_001138387.1" target="_self" alt="fibroblast growth factor receptor 2 isof"  &gt;
&lt;area shape="rect" coords="516,1554,691,1542" href="http://www.ncbi.nlm.nih.gov/protein/NP_001138390.1" target="_self" alt="fibroblast growth factor receptor 2 isof"  &gt;
&lt;area shape="rect" coords="516,1577,691,1566" href="http://www.ncbi.nlm.nih.gov/protein/NP_001138388.1" target="_self" alt="fibroblast growth factor receptor 2 isof"  &gt;
&lt;area shape="rect" coords="513,1600,695,1589" href="http://www.ncbi.nlm.nih.gov/protein/NP_002028.1" target="_self" alt="tyrosine-protein kinase Fyn isoform a [H"  &gt;
&lt;area shape="rect" coords="516,1624,691,1612" href="http://www.ncbi.nlm.nih.gov/protein/NP_075259.4" target="_self" alt="fibroblast growth factor receptor 2 isof"  &gt;
&lt;area shape="rect" coords="516,1647,691,1635" href="http://www.ncbi.nlm.nih.gov/protein/NP_001138385.1" target="_self" alt="fibroblast growth factor receptor 2 isof"  &gt;
&lt;area shape="rect" coords="516,1670,691,1658" href="http://www.ncbi.nlm.nih.gov/protein/NP_000132.3" target="_self" alt="fibroblast growth factor receptor 2 isof"  &gt;
&lt;area shape="rect" coords="452,1693,691,1682" href="http://www.ncbi.nlm.nih.gov/protein/NP_006334.2" target="_self" alt="tyrosine-protein kinase Mer precursor [H"  &gt;
&lt;area shape="rect" coords="480,1716,693,1705" href="http://www.ncbi.nlm.nih.gov/protein/NP_543013.1" target="_self" alt="tyrosine-protein kinase Srms [Homo sapie"  &gt;
&lt;area shape="rect" coords="516,1739,695,1728" href="http://www.ncbi.nlm.nih.gov/protein/NP_001229708.1" target="_self" alt="tyrosine-protein kinase Fyn isoform d [H"  &gt;
&lt;area shape="rect" coords="512,1763,688,1751" href="http://www.ncbi.nlm.nih.gov/protein/NP_000052.1" target="_self" alt="tyrosine-protein kinase BTK [Homo sapien"  &gt;
&lt;area shape="rect" coords="513,1786,688,1774" href="http://www.ncbi.nlm.nih.gov/protein/NP_001706.2" target="_self" alt="tyrosine-protein kinase Blk [Homo sapien"  &gt;
&lt;area shape="rect" coords="495,1809,691,1797" href="http://www.ncbi.nlm.nih.gov/protein/NP_004374.1" target="_self" alt="tyrosine-protein kinase CSK [Homo sapien"  &gt;
&lt;area shape="rect" coords="516,1832,695,1821" href="http://www.ncbi.nlm.nih.gov/protein/NP_694592.1" target="_self" alt="tyrosine-protein kinase Fyn isoform b [H"  &gt;
&lt;area shape="rect" coords="516,1855,705,1844" href="http://www.ncbi.nlm.nih.gov/protein/NP_075252.2" target="_self" alt="fibroblast growth factor receptor 4 isof"  &gt;
&lt;area shape="rect" coords="516,1879,705,1867" href="http://www.ncbi.nlm.nih.gov/protein/NP_002002.3" target="_self" alt="fibroblast growth factor receptor 4 isof"  &gt;
&lt;area shape="rect" coords="467,1902,687,1890" href="http://www.ncbi.nlm.nih.gov/protein/NP_002935.2" target="_self" alt="proto-oncogene tyrosine-protein kinase R"  &gt;
&lt;area shape="rect" coords="513,1925,687,1913" href="http://www.ncbi.nlm.nih.gov/protein/NP_005408.1" target="_self" alt="proto-oncogene tyrosine-protein kinase S"  &gt;
&lt;area shape="rect" coords="432,1948,697,1937" href="http://www.ncbi.nlm.nih.gov/protein/NP_000450.2" target="_self" alt="angiopoietin-1 receptor precursor [Homo "  &gt;
&lt;area shape="rect" coords="499,1971,687,1960" href="http://www.ncbi.nlm.nih.gov/protein/NP_000206.2" target="_self" alt="tyrosine-protein kinase JAK3 [Homo sapie"  &gt;
&lt;area shape="rect" coords="517,1995,688,1983" href="http://www.ncbi.nlm.nih.gov/protein/NP_005239.1" target="_self" alt="tyrosine-protein kinase Fgr [Homo sapien"  &gt;
&lt;area shape="rect" coords="495,2018,705,2006" href="http://www.ncbi.nlm.nih.gov/protein/NP_647611.1" target="_self" alt="megakaryocyte-associated tyrosine-protei"  &gt;
&lt;area shape="rect" coords="465,2041,703,2029" href="http://www.ncbi.nlm.nih.gov/protein/NP_996844.1" target="_self" alt="leukocyte tyrosine kinase receptor isofo"  &gt;
&lt;area shape="rect" coords="465,2064,703,2053" href="http://www.ncbi.nlm.nih.gov/protein/NP_002335.2" target="_self" alt="leukocyte tyrosine kinase receptor isofo"  &gt;
&lt;area shape="rect" coords="460,2087,697,2076" href="http://www.ncbi.nlm.nih.gov/protein/NP_055030.1" target="_self" alt="insulin receptor-related protein precurs"  &gt;
&lt;area shape="rect" coords="523,2111,695,2099" href="http://www.ncbi.nlm.nih.gov/protein/NP_694593.1" target="_self" alt="tyrosine-protein kinase Fyn isoform c [H"  &gt;
&lt;area shape="rect" coords="495,2134,705,2122" href="http://www.ncbi.nlm.nih.gov/protein/NP_002369.2" target="_self" alt="megakaryocyte-associated tyrosine-protei"  &gt;
&lt;area shape="rect" coords="495,2157,705,2145" href="http://www.ncbi.nlm.nih.gov/protein/NP_647612.1" target="_self" alt="megakaryocyte-associated tyrosine-protei"  &gt;
&lt;area shape="rect" coords="487,2180,691,2169" href="http://www.ncbi.nlm.nih.gov/protein/NP_006284.2" target="_self" alt="tyrosine-protein kinase receptor TYRO3 ["  &gt;
&lt;area shape="rect" coords="509,2203,688,2192" href="http://www.ncbi.nlm.nih.gov/protein/NP_001712.1" target="_self" alt="cytoplasmic tyrosine-protein kinase BMX "  &gt;
&lt;area shape="rect" coords="465,2226,702,2215" href="http://www.ncbi.nlm.nih.gov/protein/NP_001690.2" target="_self" alt="tyrosine-protein kinase receptor UFO iso"  &gt;
&lt;area shape="rect" coords="465,2250,702,2238" href="http://www.ncbi.nlm.nih.gov/protein/NP_068713.2" target="_self" alt="tyrosine-protein kinase receptor UFO iso"  &gt;
&lt;area shape="rect" coords="501,2273,709,2261" href="http://www.ncbi.nlm.nih.gov/protein/NP_000199.2" target="_self" alt="insulin receptor isoform Long preproprot"  &gt;
&lt;area shape="rect" coords="501,2296,709,2284" href="http://www.ncbi.nlm.nih.gov/protein/NP_001073285.1" target="_self" alt="insulin receptor isoform Short prepropro"  &gt;
&lt;/map&gt;</pre>
</div>
<p>or just save as an SVG file to get href attributes automatically added:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">panel</span><span class="o">.</span><span class="n">save</span><span class="p">(</span><span class="s">&#39;blast_output.svg&#39;</span><span class="p">)</span>
</pre></div>
</div>
<object data="_images/blast_output.svg" height="500px" type="image/svg+xml" width="1100px"><embed height="500px" src="_images/blast_output.svg" type="image/svg+xml" width="1100px" /></object>
</div>
<div class="section" id="extending-the-library">
<h2>Extending the library<a class="headerlink" href="#extending-the-library" title="Permalink to this headline">¶</a></h2>
<div class="section" id="creating-a-custom-graphicfeature">
<h3>Creating a custom GraphicFeature<a class="headerlink" href="#creating-a-custom-graphicfeature" title="Permalink to this headline">¶</a></h3>
<p>Subclassing from  <tt class="docutils literal"><span class="pre">biograpy.features.BaseGraphicFeature</span></tt></p>
<p><cite>TODO</cite></p>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="modindex.html" title="Global Module Index"
             >modules</a> |</li>
        <li><a href="index.html">BioGraPy v1.0 beta documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
      &copy; Copyright 2011, Andrea Pierleoni.
      Last updated on Sep 08, 2011.
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 0.6.3.
    </div>
  </body>
</html>